pkg/controller/vitesskeyspace/reconcile_resharding.go

```diff
@@ -13,25 +15,69 @@ import (
 	"vitess.io/vitess/go/vt/wrangler"
 )
 
+const (
+	maxSafeVReplicationLag = 10
+)
+
 func (r *ReconcileVitessKeyspace) reconcileResharding(ctx context.Context, vtk *planetscalev2.VitessKeyspace) error {
 	wr, ts, err := newWrangler(ctx, vtk.Spec.GlobalLockserver)
 	if err != nil {
 		return err
 	}
 	defer ts.Close()
 
-	workflows, err := wr.ListAllWorkflows(ctx, vtk.Spec.Name)
+	workflowList, err := wr.ListAllWorkflows(ctx, vtk.Spec.Name)
 	if err != nil {
-		workflows = make([]string, 0)
+		workflowList = make([]string, 0)
 	}
 
-	vtk.Status.ActiveWorkflows = workflows
-	// TODO: Address when ConditionUnknown is applicable.
-	if len(workflows) != 0 {
-		vtk.Status.ReshardingInProgress = corev1.ConditionTrue
-	} else {
-		vtk.Status.ReshardingInProgress = corev1.ConditionFalse
+	reshardingInProgress := corev1.ConditionUnknown
+	workflows := make([]planetscalev2.WorkflowStatus, 0, len(workflowList))
+	for _, workflowName := range workflowList {
+		workflow, err := wr.ShowWorkflow(ctx, workflowName, vtk.Spec.Name)
+		if err != nil {
+			return err
+		}
+		workflowStatus := planetscalev2.WorkflowStatus{
+			Workflow: workflow.Workflow,
+		}
+
+		for name, status := range workflow.ShardStatuses {
+			if status.MasterIsServing {
+				shard := strings.Split(name, "/")[0]
+				vtk.Status.ServingShards = append(vtk.Status.ServingShards, shard)
+			}
+			for _, vReplRow := range status.MasterReplicationStatuses {
+				if vReplRow.State == "Error" {
```

Comment:
We should be able to extract the error message, right? We could use that in the `message` field of a status condition for resharding, e.g. to explain why the `ReshardingInSync` condition is `False`.

We'd have to be careful though to avoid spurious updates. Are the entries in MasterReplicationStatuses always in the same order? We also need to decide if we show all errors, or only the first row with an error.

Reply:
Yes, the error message will be in the `Message` field of the `ReplicationStatus` (which really represents a vrepl row). The same one that we are looking at the State field on. 

I think for simplicity's sake we can just show the first error for now. We can always expand that in the future to include all errors.

Reply:
To answer your question on ordering - my guess is no. It's the result of a SQL query with no `ORDER BY`, so I don't think order can be guaranteed.

Reply:
Showing the first error sounds good for now. We just need to make sure to iterate in deterministic order so we don't pick different "first errors" for the same actual data set.

Reply:
It sounds like for that to be true we have to collect the errors regardless, sort them and then return the first?

Reply:
Decided to go with the idea of summing up errors, sorting and returning the first of a sorted list because it seemed like a much simpler idea.